tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/TensorFlow.java

```diff
@@ -138,4 +150,82 @@ private TensorFlow() {}
       throw e;
     }
   }
+
+  // to keep them from getting GC'd
+  private static final Set<GradFunc> gradientFuncs =
```

Comment:
Looks like all the methods that access this are synchronized. Maybe a note here to say that that is required for correctness?

Reply:
I mean that the doc should say something like `accesses to this set should be synchronized on this class`. Because all the methods that use it are synchronized and if they weren't then we could get odd races.